Graphics/Slicer/Machine/Infill.hs

```diff
@@ -80,65 +80,68 @@ infillLineSegInside contour childContours line
           xMax = maximum $ xOf <$> contourPoints
-          yMax = maximum $ yOf <$> contourPoints
-          xMaxOutside = xMax + ls
-          yMaxOutside = yMax + ls
-          lsX = sqrt $ ls*ls+ls*ls
+          -- line spacing, taking into account the slope.
+          lss = sqrt $ ls*ls+ls*ls
 
 -- Generate lines over entire print area, where each one is aligned with a +1 slope.
--- FIXME: other ways to only generate covering lines over the given contour?
 -- FIXME: assumes we're in positive space.
-coveringLineSegsPositive :: Contour -> ℝ -> [LineSeg]
-coveringLineSegsPositive (PointSequence contourPoints) ls = flip LineSeg s . f <$> [0,lsY..yMax + xMax]
-    where s =  Point2 (xMaxOutside + yMaxOutside,- xMaxOutside - yMaxOutside)
-          f v = Point2 (0,v)
-          xMax = maximum $ xOf <$> contourPoints
+coveringLineSegsPositive :: Contour -> ℝ -> [PLine2]
+coveringLineSegsPositive (PointSequence contourPoints) ls = eToPLine2 . flip LineSeg slope . f <$> [0,lss..(xMax-xMin)+(yMax-yMin)+lss]
+    where slope =  Point2 (1,-1)
+          f v = Point2 (v+yDiff,0)
+          yDiff = (xMin+yMin)
+          yMinRaw = minimum $ yOf <$> contourPoints
+          yMin = head $ filter (> yMinRaw) [0, lss..]
           yMax = maximum $ yOf <$> contourPoints
-          xMaxOutside = xMax + ls
-          yMaxOutside = yMax + ls
-          lsY = sqrt $ ls*ls+ls*ls
+          xMin = minimum $ xOf <$> contourPoints
+          xMax = maximum $ xOf <$> contourPoints
+          -- line spacing, taking into account the slope.
+          lss = sqrt $ ls*ls+ls*ls
 
 -- Generate lines covering the entire contour, where each line is aligned with the Y axis.
 -- FIXME: assumes we're in positive space.
-coveringLineSegsVertical :: Contour -> ℝ -> [LineSeg]
-coveringLineSegsVertical (PointSequence contourPoints) ls = flip LineSeg s . f <$> [xMin-ls,xMin..xMax]
-    where s =  Point2 (0,yMax+ls)
-          f v = Point2 (v,-ls)
+coveringLineSegsVertical :: Contour -> ℝ -> [PLine2]
+coveringLineSegsVertical (PointSequence contourPoints) ls = eToPLine2 . flip LineSeg slope . f <$> [xMin,xMin+ls..xMax]
+    where slope =  Point2 (0,1)
+          f v = Point2 (v,0)
           xMinRaw = minimum $ xOf <$> contourPoints
-          xMin = head $ filter (> xMinRaw) [-ls, 0..]
+          xMin = head $ filter (> xMinRaw) [0, ls..]
           xMax = maximum $ xOf <$> contourPoints
-          yMax = maximum $ yOf <$> contourPoints
 
 -- Generate lines covering the entire contour, where each line is aligned with the X axis.
 -- FIXME: assumes we're in positive space.
-coveringLineSegsHorizontal :: Contour -> ℝ -> [LineSeg]
-coveringLineSegsHorizontal (PointSequence contourPoints) ls = flip LineSeg s . f <$> [yMin-ls,yMin..yMax]
-    where s =  Point2 (xMax+ls,0)
-          f v = Point2 (-ls,v)
+coveringLineSegsHorizontal :: Contour -> ℝ -> [PLine2]
+coveringLineSegsHorizontal (PointSequence contourPoints) ls = eToPLine2 . flip LineSeg slope . f <$> [yMin,yMin+ls..yMax]
```

Comment:
This and `coveringLineSegsVertical` could be unified into one function, and then partially applied to avoid the code duplication.

Reply:
Disagree. these four functions do the same thing, at different angles. eventually, there will be even more complicated functions for this stuff. these are the 'simple placeholders'.